mantis-examples/mantis-examples-sine-function/src/main/java/io/mantisrx/mantis/examples/sinefunction/SineFunction.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.mantis.examples.sinefunction;
+
+import static io.mantisrx.mantis.examples.sinefunction.SineFunctionJob.INTERVAL_SEC;
+import static io.mantisrx.mantis.examples.sinefunction.SineFunctionJob.RANDOM_RATE;
+import static io.mantisrx.mantis.examples.sinefunction.SineFunctionJob.RANGE_MAX;
+import static io.mantisrx.mantis.examples.sinefunction.SineFunctionJob.RANGE_MIN;
+import static io.mantisrx.mantis.examples.sinefunction.SineFunctionJob.USE_RANDOM_FLAG;
+
+import io.mantisrx.mantis.examples.sinefunction.core.Point;
+import io.mantisrx.runtime.core.MantisStream;
+import io.mantisrx.runtime.core.MantisStreamImpl;
+import io.mantisrx.runtime.core.functions.ReduceFunctionImpl;
+import io.mantisrx.runtime.core.sinks.ObservableSinkImpl;
+import io.mantisrx.runtime.core.sources.ObservableSourceImpl;
+import io.mantisrx.runtime.parameter.type.BooleanParameter;
+import io.mantisrx.runtime.parameter.type.DoubleParameter;
+import io.mantisrx.runtime.parameter.type.IntParameter;
+import io.mantisrx.runtime.parameter.validator.Validators;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class SineFunction {
+    public static void main(String[] args) {
+        final double amplitude = 5.0;
+        final double frequency = 1;
+        final double phase = 0.0;
+
+        MantisStreamImpl.init()
```

Comment:
IMO, the chain below looks cleaner. Also, not very sure if we should mix MantisStreams with the parameter definition itself. IMO, it should be part of the job definition for the time being. We should think about getting rid of it in the future. 
```
MantisStreamFactory
  .source(SourceFunction)
  .filter()
  .map()
  .keyBy()
  .tumblingWindow()
  .reduce()
  .sink() 
```

Reply:
yeah, agree decoupling the parameter definition from the stream building is cleaner.